code/modules/mob/living/carbon/alien/special/facehugger.dm

```diff
@@ -134,7 +134,9 @@
 		// gotta have a head to be implanted (no changelings or sentient plants)
 		if(!target.get_bodypart(BODY_ZONE_HEAD))
 			return FALSE
-
+		// gotta be able to have the xeno implanted
+		if(HAS_TRAIT(M, TRAIT_XENO_IMMUNE))
+			return FALSE
 		if(target.getorgan(/obj/item/organ/alien/hivenode) || target.getorgan(/obj/item/organ/body_egg/alien_embryo))
 			return FALSE

```

Comment:
I would love it if you could kill these two lines by also making them part of the trait check.

Reply:
Which two lines? The xeno organ check or the bodypart check?

Reply:
140 and 141

Reply:
I'll try that and test it tomorrow when I've got some time

Reply:
@nemvar changed it so that the insertion of both hivenode and embryo adds the trait, and removes the trait when those organs are removed.